tfjs-backend-wasm/src/cc/kernels/ArgMin.cc

```diff
@@ -0,0 +1,76 @@
+/* Copyright 2023 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ===========================================================================*/
+
+#ifdef __EMSCRIPTEN__
+#include <emscripten.h>
+#endif
+
+#include <cstddef>
+
+#include "tfjs-backend-wasm/src/cc/backend.h"
+#include "tfjs-backend-wasm/src/cc/util.h"
+
+namespace tfjs::wasm {
+
+namespace {
+
+template <typename T>
+void ArgMinImpl(const T* x, const int outer_size, const int inner_size,
+                int32_t* out_buf) {
+  for (int i = 0; i < outer_size; ++i) {
+    const int offset = i * inner_size;
+    T min_value = x[offset];
+    int min_index = 0;
```

Comment:
Maybe we can factor out some of this array indexing code. It seems like several ops act on just the last dimension of the tensor. Could we have a function that applies another function to the last dimension (or maybe a more general one that works on the last n dimensions)?

Then, we could just pass it this function.

```c
void ArgMaxInner(const T* x, const int size, int32_t* out_buf ) {
  T min_value = x[0];
  int min_index = 0;
  for (int i = 1; i < size; i++) {
    t Value = x[i];
    if (value < min_value) {
      min_value = value;
      min_index = i;
  }
  out_buf[i] = min_index;
}
```

Probably better for a later PR, though.

Reply:
Added